src/Case-TGV.f90

```diff
@@ -412,6 +375,101 @@ subroutine postprocess_tgv(ux1,uy1,uz1,phi1,ep1)
           write(42,'(20e20.12)') (itime-1)*dt,eek,eps,eps2,enst
           flush(42)
        endif
+
+       if(mhd_active) then
+
+          !SPATIALLY-AVERAGING
+          temp1=zero
+          temp2=zero
+          temp3=zero
+          do k=1,xsize3
+             do j=1,xsize2
+                do i=1,xsize1
+                   temp1=temp1+zpfive*((Bm(i,j,k,1))**2+(Bm(i,j,k,2))**2+(Bm(i,j,k,3))**2)
+                   temp2=temp2+zpfive*Je(i,j,k,1)**2+Je(i,j,k,2)**2+Je(i,j,k,3)**2
+                enddo
+             enddo
+          enddo
+          call MPI_ALLREDUCE(temp1,   eem,1,real_type,MPI_SUM,MPI_COMM_WORLD,code)
```

Comment:
temporary variables can be avoided when performing MPI reduction in place

Reply:
As comment before